### PR TITLE
Expose boards as calendars

### PR DESCRIPTION
**Pull request by juliushaertl**
Implements https://github.com/nextcloud/deck/issues/15

- [x] Requires https://github.com/nextcloud/server/pull/19196 so 19+ only change
- [x] We could switch directly to VTODO entries if https://github.com/nextcloud/calendar/pull/1798 is ready
- [ ] Open deck from the calendar app https://github.com/nextcloud/calendar/pull/2478

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
Will Deck then also move to the CalDAV interface / become a CalDAV client?

**Comment by juliushaertl** [comment]
No, until now this is only for exposing cards with duedates to the calendar/tasks app, related to the discussion in  https://github.com/nextcloud/deck/pull/1319#issuecomment-591388534

File: lib/DAV/Calendar.php [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+				switch ($key) {
+					case '{DAV:}displayname':
+						if (mb_substr($value, 0, strlen('Deck: '))) {
+							$value = mb_substr($value, strlen('Deck: '));
+						}
+						$this->board->setTitle($value);
+						break;
+					case '{http://apple.com/ns/ical/}calendar-color':
+						$this->board->setColor(substr($value, 1));
+						break;
```
Comment (juliushaertl) [inline lib/DAV/Calendar.php:181]: @georgehrke Is there any standard where the format of this is specified? Are there any other values than hex colors that could occur here?

Comment (juliushaertl) [reply lib/DAV/Calendar.php:181]: ping :wink:

Comment (georgehrke) [reply lib/DAV/Calendar.php:181]: `{http://apple.com/ns/ical/}calendar-color` always contains a hex color. Usually it's `#rrggbb`. macOS calendar is sending `#rrggbbaa` though, so you would have to filter that out.

**Review by jakobroehrl — COMMENTED** [review]
It's working fine :)
But I saw the due hour is +1h in the calendar. A 12:00 due card is in the calendar show as a 13:00 due event

**Comment by hitam4450** [comment]
Maybe timezone problem ?

**Comment by juliushaertl** [comment]
Hm, I didn't have this when testing, but I pushed a commit that should force UTC interpretation for the duedate so that the clients can take care of the timezone. Could you check if that fixes it for you?

**Comment by jakobroehrl** [comment]
> Hm, I didn't have this when testing, but I pushed a commit that should force UTC interpretation for the duedate so that the clients can take care of the timezone. Could you check if that fixes it for you?

Now it's working.

**Comment by juliushaertl** [comment]
Let's get this in for the next beta :rocket:

**Pull request merged**